src/Core/MailTemplate/FolderThemeCatalog.php

```diff
@@ -0,0 +1,280 @@
+    }
+
+    /**
+     * Returns the list of found themes (non empty folders, in the mail themes
+     * folder).
+     *
+     * @throws FileNotFoundException
+     * @throws TypeException
+     *
+     * @return ThemeCollectionInterface
+     */
+    public function listThemes()
+    {
+        $this->checkThemesFolder();
+
+        $finder = new Finder();
+        $finder->directories()->in($this->mailThemesFolder)->depth(0);
+        $mailThemes = new ThemeCollection();
+        /** @var SplFileInfo $mailThemeFolder */
+        foreach ($finder as $mailThemeFolder) {
+            $dirFinder = new Finder();
+            $dirFinder->files()->in($mailThemeFolder->getRealPath());
+            if ($dirFinder->count() > 0) {
+                $mailTheme = new Theme($mailThemeFolder->getFilename());
+                $layouts = $this->listLayouts($mailTheme->getName());
+                $mailTheme->setLayouts($layouts);
+                $mailThemes[] = $mailTheme;
+            }
+        }
+
+        //This hook allows you to add/remove a mail theme
+        $this->hookDispatcher->dispatchWithParameters(
+            ThemeCatalogInterface::LIST_MAIL_THEMES_HOOK,
+            ['mailThemes' => $mailThemes]
+        );
+
+        return $mailThemes;
+    }
+
+    /**
+     * @param string $theme
+     *
+     * @throws FileNotFoundException
+     * @throws InvalidArgumentException
+     * @throws TypeException
+     *
+     * @return ThemeInterface
+     */
+    public function getByName($theme)
+    {
+        /** @var ThemeCollectionInterface $availableThemes */
+        $availableThemes = $this->listThemes();
+        $themeNames = [];
+        /** @var ThemeInterface $availableTheme */
+        foreach ($availableThemes as $availableTheme) {
+            if ($theme === $availableTheme->getName()) {
+                return $availableTheme;
+            }
+            $themeNames[] = $availableTheme->getName();
+        }
+
+        throw new InvalidArgumentException(sprintf(
+            'Invalid requested theme "%s", only available themes are: %s',
+            $theme,
+            implode(', ', $themeNames)
+        ));
+    }
+
+    /**
+     * @param string $mailTheme
+     *
+     * @throws FileNotFoundException
+     * @throws TypeException
+     *
+     * @return LayoutCollectionInterface
+     */
+    private function listLayouts($mailTheme)
+    {
+        $mailThemeFolder = implode(DIRECTORY_SEPARATOR, [$this->mailThemesFolder, $mailTheme]);
+        //This hook allows to change the mail them folder
+        $this->hookDispatcher->dispatchWithParameters(
+            static::GET_MAIL_THEME_FOLDER_HOOK,
+            [
+                'mailTheme' => $mailTheme,
+                'mailThemeFolder' => &$mailThemeFolder,
+            ]
+        );
+        $this->checkThemeFolder($mailThemeFolder);
+
+        $mailThemeLayouts = new LayoutCollection();
+        $this->listCoreLayouts($mailThemeLayouts, $mailThemeFolder);
```

Comment:
Excepts these are not getter, these are methods which search for layouts and populate the collection. This responsibility should not be in the collection (too many dependency to do this job).
Besides today the layouts are found with `FolderThemeCatalog.php` but in the future it will probably be stored in database, so we will have a new `DatabaseThemeCatalog` doing this job and the collection won't change.